FallFestGUIClient/src/com/ibm/sk/ff/gui/client/GUIFacade.java

```diff
@@ -123,39 +128,49 @@ public void remove(final GUIObject data) {
 
 	public void remove(final GUIObject[] data) {
 		if (data.length > 0) {
-			this.CLIENT.postMessage(GUIOperations.REMOVE.toString() + "/" + data[0].getType().toString(), Mapper.INSTANCE.pojoToJson(data));
+			this.CLIENT.postMessage(REMOVE.toString() + "/" + data[0].getType().toString(), INSTANCE.pojoToJson(data));
+			steps.add(new Step(REMOVE, data));
 		}
 	}
 
 	public void showScore(final ScoreData data) {
-		this.CLIENT.postMessage(GUIOperations.SCORE.toString(), Mapper.INSTANCE.pojoToJson(data));
+		String scoreDataJson = INSTANCE.pojoToJson(data);
+		this.CLIENT.postMessage(SCORE.toString(), scoreDataJson);
 	}
 
 	//
 	// Operations
 	//
 	public void showInitMenu(final InitMenuData data) {
-		this.CLIENT.postMessage(GUIOperations.SHOW_INIT_MENU.toString(), Mapper.INSTANCE.pojoToJson(data));
+		String initMenuDataJson = INSTANCE.pojoToJson(data);
+		this.CLIENT.postMessage(SHOW_INIT_MENU.toString(), initMenuDataJson);
 	}
 
 	public void createGame(final CreateGameData data) {
-		this.CLIENT.postMessage(GUIOperations.CREATE_GAME.toString(), Mapper.INSTANCE.pojoToJson(data));
+		String createGameDataJson = INSTANCE.pojoToJson(data);
+		this.CLIENT.postMessage(CREATE_GAME.toString(), createGameDataJson);
+		steps = new ArrayList<>();
 	}
 
 	public void showResult(final ResultData data) {
-		this.CLIENT.postMessage(GUIOperations.SHOW_RESULT.toString(), Mapper.INSTANCE.pojoToJson(data));
+		String resultDataJson = INSTANCE.pojoToJson(data);
+		this.CLIENT.postMessage(SHOW_RESULT.toString(), resultDataJson);
+		steps.add(new Step(SHOW_RESULT, resultDataJson));
 	}
 
 	public void close(final CloseData data) {
-		this.CLIENT.postMessage(GUIOperations.CLOSE.toString(), Mapper.INSTANCE.pojoToJson(data));
+		String closeDataJson = INSTANCE.pojoToJson(data);
+		this.CLIENT.postMessage(CLOSE.toString(), closeDataJson);
+		write(steps);
 	}
 
 	// Poll for events
 	private void checkEvents() {
-		final String events = this.CLIENT.getMessage(GUIOperations.EVENT_POLL.toString());
+		final String events = this.CLIENT.getMessage(EVENT_POLL.toString());
 		if (events != null && events.length() > 0) {
-			final GuiEvent[] swp = Mapper.INSTANCE.jsonToPojo(events, GuiEvent[].class);
+			final GuiEvent[] swp = INSTANCE.jsonToPojo(events, GuiEvent[].class);
 			castEvent(swp);
+			steps.add(new Step(EVENT_POLL, swp));
```

Comment:
Event poll asi nechceme ukladat. Ten je aj tak iba v Menu, cize s hrou moc toho nema spolocneho. Ci to mas kvoli hracom?